### DISCUSSION TITLE
How to forward all the request to another computer on the same Network

### DISCUSSION THREAD

**Discussion by SpinB** [DNSCrypt/dnscrypt-proxy#2181] [Q&A] [answered]
Hello,

How do I set up a computer using DNSCrypto-proxy (acting as a client), not the windows DNS settings, to force all outgoing queries to another computer on the same network using DNSCrypto-proxy (acting as a server).

Thanks,
Alex

**Comment by jedisct1** [comment, marked as answer]
Are you trying to run your own DNSCrypt server?

In that case, you need to install [encrypted dns server](https://github.com/dnscrypt/encrypted-dns-server), that now also works on Windows.

**Reply by SpinB** [reply]
No, I have 2 Laptops, Laptop A (Server/10.10.0.5) with DNSCrypt-proxy installed and listening to 10.10.0.5:53 and Laptop B (Client/10.10.0.10) with the network card DNS settings set to 10.10.0.5 and everything works fine, what I want to do is install DNSCrypt-proxy on Laptop B and forward all the requests to Laptop A using the DNSCrypt-proxy instance, instead of using the windows UDP and have DNSCrypt-proxy TCP, something like having the network DNS settings 127.0.0.1 and the TOML file configured to make all the queries to Laptop A.
It's possible to do this?

**Reply by jedisct1** [reply]
This is a bit convoluted, but the way to do it would be to indeed run a DNSCrypt server on Laptop A, that forwards to `10.10.0.5:53`, and use that server's stamp on Laptop B.

**Reply by SpinB** [reply]
Sorry, but can you please explain to me, how I configure the TOML file on Laptop B, do I do something like this on the [anonymized_dns]:

routes = [
   { server_name='*', via=['*Laptop A server stamp']}
]

**Comment by lifenjoiner** [comment]
> Laptop A (Server/10.10.0.5) with DNSCrypt-proxy installed and listening to 10.10.0.5:53 and Laptop B (Client/10.10.0.10) with the network card DNS settings set to 10.10.0.5 and everything works fine

I think it is the best practice, for one entry.

What is the advantage of "install DNSCrypt-proxy on Laptop B and forward all the requests to Laptop A using the DNSCrypt-proxy instance"? If you prefer 2 instances, why aren't they 2 standalone?

**Reply by SpinB** [reply]
Well, I guess if there's no solution for what I want to do, I will have to run DNSCrypt-proxy standalone when I'm using a wireless connection on Laptop B then.
